Refactor login screen

won0104 opened the pull request:
## 💡 Issue number (ex. #1000)
#42 
## 🚀 Description
- [x] 기본 컴포넌트를 `glue-stack-ui`변경
- [x] 디렉토리 분류하기
- [x] 스타일, 타입, 상수 분리 하기 및 중복 코드 제거

## 💻 etc.
제가 처음해서 살짝 커밋 로그가 많이 두서없고 지저분하네요,,, 
다음 리펙토링은 커밋을 좀 크게 크게 나누고, 커밋 본문 내용을 작성해보겠습니다!!ㅎㅎ,,

결론적으로 두가지 부분을 조금 다르게 수정한 것 같습니다. (로직은 너무 잘 나눠져있어서 따로 분리할 것이 없었습니다👍)

1. style에서 반복되는 부분은 파일 내 전역변수로 사용하였습니다.
  - 저는 이 방법이 중복 코드도 줄고, 혹시 나중에 디자인적 요소에서 변경이 필요하다면 수정이 편할 것 같아 저의 취향대로 수정을 했는데 다른분들은 어떻게 생각하실지 궁금합니다!

2. `glue-stack-ui`의 경우 import 할 때 하나씩 import하는게 지저분해 보여 `components/ui/index.ts` 에 자주 사용하는 컴포넌트들을 먼저 import를 한 후, 사용하는 본문에서는` import {Box, VStack, HStack, Text} from '@/components/ui/index.ts'` 방법과 같이 한번에 import 할 수 있도록 하였습니다! 다른 분들도 사용하시면 좋을 것 같아요~

+) `Loder`를 `LoginForm`으로 옮겨서 `props`를 사용하지 않으려고 했는데 이게 `LoginForm` 크기 때문에 로더가 반만 뜨는 사태가 발생했습니다. style을 좀 고치면 될 것 같아 수정을 해보았으나 제가 짠 코드가 아니라 그런지 쉽지 않아서 결국 다시 `LoginScreen`단으로 옮겼네요..하하ㅜㅜ 

+) 저번주에 시작하기로 했는데 생각보다 토익에 시간을 뺏겨서 많이 진행하지 못했네요...ㅠㅠ 이번주에는 좀 더 열심히 참여하도록 하겠습니다..!!!

won0104 committed:
refactor : LoginScreen.tsx 파일 삭제

won0104 committed:
refactor: LoginScreenStyle.ts 파일 생성

won0104 committed:
refactor: index.tsx 파일 생성, GRADIENT 상수 사용

won0104 committed:
refactor: 안쓰는 매개변수 삭제

won0104 committed:
refactor: LoginForm.tsx 필요 없는 파일 삭제

won0104 committed:
refactor: 경로 수정

won0104 committed:
refactor: 경로 수정

won0104 committed:
refactor: LoginForm Props 분리

won0104 committed:
refactor: gluestack 변경

won0104 committed:
refactor: Style 중복 코드 제거

won0104 committed:
refactor: Gluestack export 합치기

won0104 committed:
refactor: 전역 변수 설정

won0104 committed:
refactor: LoginScreen에서 LoginForm으로 loder 이동

won0104 committed:
refactor: LoginScreen에서 LoginForm으로 loder 이동

won0104 committed:
refactor: loader loginScreen단에서 호출

KIMJINWOO4 commented on src/components/LoginFrom/LoginFromStyle.ts:
```diff
@@ -0,0 +1,106 @@
+const subTextColor = '#5D5D5D';
+const pointColor = '#009499';
+
+const commonTextStyle = {
```
이렇게 파일 단위로 관리해주니 좋은 것 같습니다.
전체적으로 이런 방향을 잡아도 좋을 것 같아용

KIMJINWOO4 commented on src/components/LoginFrom/index.tsx:
```diff
@@ -0,0 +1,143 @@
+                    <TextInput
+                        style={LoginFormStyles.input}
+                        placeholder="이메일을 입력해주세요."
+                        placeholderTextColor="#B9B9B9"
```
`placeholder`와 `placeholderTextColor`도 `constant` 폴더에서 관리해주면 좋을 것 같아요 ~

KIMJINWOO4 commented on src/components/LoginFrom/index.tsx:
```diff
@@ -0,0 +1,143 @@
+                        placeholderTextColor="#B9B9B9"
+                        value={email}
+                        onChangeText={setEmail}
+                        returnKeyType="next"
```
이 부분도 `constant` 폴더로 옮겨주면 좋을 것 같습니다 !

KIMJINWOO4 commented on src/components/LoginFrom/index.tsx:
```diff
@@ -0,0 +1,143 @@
+                        style={LoginFormStyles.findTextContainer}
+                        onPress={() =>
+                            navigation.navigate('FindPassword', {
+                                title: '비밀번호 찾기',
```
이 부분도 상수 관리 방식을 일관성있게 하면 좋을 듯 합니다..!!

KIMJINWOO4 commented on src/screens/LoginScreen/index.tsx:
```diff
@@ -0,0 +1,41 @@
+                }
+                enableOnAndroid={true}
+                extraScrollHeight={20}
+                keyboardShouldPersistTaps="handled"
```
이 부분도 `true`, `20` 과 같은 부분은 상수 관리를 일관적으로 해주면 좋을 것 같아요 `keyboardShouldPersistTaps`는 변수에 넣으면 안되긴 하더라구요 !!

KIMJINWOO4 reviewed (commented):
와우 ! 대 채 원의 첫 번째 PR 감격스럽네용..
첫 번째임에도 규칙을 거의 다 지켜주셔서 감사합니다.
몇가지 수정사항만 체크해뒀으니 참고하시면 좋을 듯 해용
그리고 `LoginForm`의 크기가 이렇게 완성되고 보니 너무 커서 가독성이 아주 쪼끔 떨어진다는 생각이 드네요. 
분리 단위에 대해 한 번 더 같이 고민해보면 좋을 것 같아요 넘 수고하셨슴당

KIMJINWOO4 commented on components/ui/index.ts:
```diff
@@ -0,0 +1,4 @@
+export {Box} from './box'
+export {HStack} from './hstack'
+export {VStack} from './vstack'
+export {Text} from './text'
```
이렇게 관리하니 더욱 좋은 것 같습니다 !!

won0104 committed:
refactor: LoginForm에서 inputFild 분리

중복 코드를 줄이고자 LoginFrom에서 inputFild 컴포넌트를 분리했습니다.
디자인을 확인해보니 이 부분은 따로 사용하지 않을 것 같아 LoginForm 밑에 디렉토리 추가해서 사용했습니다.

won0104 committed:
refactor: LoginForm 상수 관리 방식 변경

taeyomi reviewed (commented):
우왕 수고하셨습니다~~!!
